specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/codeVersion.json

```diff
@@ -0,0 +1,400 @@
+          "Create or Update CodeVersion": {
+            "$ref": "./examples/createOrUpdateCodeVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "machineLearningServices.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeVersionNameParameter"
+          },
+          {
+            "$ref": "#/parameters/CodeVersionVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Code entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "CodeVersionNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Name for CodeVersion.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "CodeVersionVersionParameter": {
+      "in": "path",
+      "name": "version",
+      "type": "string",
+      "description": "Version for CodeVersion.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "CodeVersionVersionOptionalParameter": {
+      "in": "query",
+      "name": "version",
+      "type": "string",
+      "description": "Version for CodeVersion.",
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "AssetPath": {
+      "description": "Details of an AssetUri.",
+      "type": "object",
+      "properties": {
+        "path": {
+          "description": "The path of file/directory.",
+          "type": "string"
+        },
+        "isDirectory": {
+          "description": "Whether the path defines a directory or a single file.",
+          "type": "boolean"
+        }
+      },
+      "additionalProperties": false
+    },
+    "CodeVersion": {
+      "description": "Details of an RegisteredCode",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
remove or rename.  There is a "name" property at the top level which should indicate the resources name.  The name of the code container is also already present in the URI

Reply:
should these sort of duplicates show up in the top-most level field or the bottom-most level field?